vaadin-context-menu-flow-parent/vaadin-context-menu-flow/src/main/java/com/vaadin/flow/component/contextmenu/ContextMenuBase.java

```diff
@@ -194,6 +200,7 @@ public boolean isOpenOnClick() {
      * Closes this context menu if it is currently open.
      */
     public void close() {
+        closedProgrammatically = true;
         getElement().callJsFunction("close");
```

Comment:
```suggestion
        getElement().setProperty("opened", false);
        getElement().callJsFunction("close");
```
This seems to work as well and doesn't require adding state.

Reply:
Thanks, tested it and it does work. Updated the PR.